### ISSUE TITLE
GLIBC versions: 2.32/2.34 is required, while container has 2.31

### ISSUE THREAD

**Issue by leguaan** [mathinstitut/goemaxima#7]
Hello, we've been trying to use 2022082900, but we're getting this error:

```
/opt/maxima/bin/goweb: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/maxima/bin/goweb)
/opt/maxima/bin/goweb: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/maxima/bin/goweb)
```

Checking the glibc version, it reveals that the resulting image is using glibc 2.31.
```
root@a675fe2e6c4d:/# ldd --version
ldd (Debian GLIBC 2.31-13+deb11u5) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

I've built the image on Arch Linux which usually has a quite late glibc installed, could this cause an issue? The host machine uses glibc 2.36.

**Comment by leguaan** [comment]
I have now built the image inside the `debian:11` container, which seems to have worked. It is now running and responding as expected.

```
local: # docker run -it -v $PWD/goemaxima:/goemaxima -v /var/run/docker.sock:/var/run/docker.sock debian:11 bash
container: # apt update
container: # apt install -y golang docker.io
container: # bash build.sh
container: # docker login [[registry]]
container: # bash buildimage.sh 2022082900 [[registry]]/[[registry namespace]]
```

PS: Please upload a new image on hub.docker.com :)

**Comment by Lennart-Kramer** [comment]
I have now set `CGO_ENABLED=0` in the `buildweb.sh` which should make it a static binary that does not rely on the OS libc.

On what image did the issue occur? Because I cannot reproduce it on the upstream images but you are asking for new images on hub.docker.com, or is that because of another issue?

**Comment by leguaan** [comment]
Hello,

Sorry I caused some confusion mentioning the Docker Hub. When I checked
Docker Hub initially it showed me the 2020xxxx00 images as being the latest
ones. I checked the tags as tried to use tag 2022082900-latest which told
me no image with such tag was available. Now it appears to me that at some
point the -latest was swapped out in favour of -dev, and looking at the
Docker Hub with a rested head I can see the 2022082900-dev tag present in
the Docker Hub.

Long ramble, but in short - you can ignore this :)

Thank you for making the static binary change, hopefully it didnt increase
the compiled file size too much.

**Comment by Lennart-Kramer** [comment]
> Sorry I caused some confusion mentioning the Docker Hub. When I checked
Docker Hub initially it showed me the 2020xxxx00 images as being the latest
ones. I checked the tags as tried to use tag 2022082900-latest which told
me no image with such tag was available. Now it appears to me that at some
point the -latest was swapped out in favour of -dev, and looking at the
Docker Hub with a rested head I can see the 2022082900-dev tag present in
the Docker Hub.
Ah no, don't use `-dev` images, those are just the builds of the latest github commit (if the CI was working at the moment, ~which it does not~), the `-latest` images are the latest tagged release.

However, `mathinstitut/goemaxima:2022082900-latest` should exist, I just tried on multiple machines.
